src/wazuh_modules/task_manager/wm_task_manager.c

```diff
@@ -267,6 +328,56 @@ cJSON* wm_task_manager_analyze_task(const cJSON *task_object, int *error_code) {
     return response;
 }
 
+cJSON* wm_task_manager_build_response_result(cJSON *res, const char *module, const char *command, char *status, int create_time, int last_update_time, char *request_command) {
+    
+    if (module != NULL) {
+        cJSON_AddStringToObject(res, json_keys[WM_TASK_MODULE], module);
+    }
+    if (command != NULL) {
+        cJSON_AddStringToObject(res, json_keys[WM_TASK_COMMAND], command);
+    }
+    if (status != NULL) {
+
+        if (!strcmp(commands_list[WM_TASK_UPGRADE_RESULT], request_command)){
+            
+            if (!strcmp(task_statuses[WM_TASK_DONE], status)){
+                cJSON_AddStringToObject(res, json_keys[WM_TASK_STATUS], upgrade_statuses[WM_TASK_UPGRADE_UPDATED]);
+            } else if (!strcmp(task_statuses[WM_TASK_IN_PROGRESS], status)){
+                cJSON_AddStringToObject(res, json_keys[WM_TASK_STATUS], upgrade_statuses[WM_TASK_UPGRADE_UPDATING]);
+            } else if (!strcmp(task_statuses[WM_TASK_FAILED], status)){
+                cJSON_AddStringToObject(res, json_keys[WM_TASK_STATUS], upgrade_statuses[WM_TASK_UPGRADE_ERROR]);
+            } else if (!strcmp(task_statuses[WM_TASK_NEW], status)){
+                cJSON_AddStringToObject(res, json_keys[WM_TASK_STATUS], upgrade_statuses[WM_TASK_UPGRADE_OUTDATED]);
+            }
+        } else {
+            cJSON_AddStringToObject(res, json_keys[WM_TASK_STATUS], status);
+        }
+    }
+    if (create_time != OS_INVALID) {
+        char timestamp[40];
+        time_t tmp = create_time;
+        struct tm tm = { .tm_sec = 0 };
+        localtime_r(&tmp, &tm);
+        strftime(timestamp, sizeof(timestamp), "%Y-%m-%d %H:%M:%S", &tm);
+        cJSON_AddStringToObject(res, json_keys[WM_TASK_CREATE_TIME], timestamp);
+    }
+    if (last_update_time != OS_INVALID) {
+        if (last_update_time > 0) {
+            char timestamp[40];
```

Comment:
timestamp generation is repeated, maybe can be done in a static method? I thnik that also the shared method `w_get_timestamp` could be used here